src/resolvers/dns.rs

```diff
@@ -878,6 +1002,96 @@ mod test {
         assert_eq!(backends.len(), 3);
     }
 
+    // The tests of our test helpers to ensure it behaves the way it claims to
+    // (which subsequent tests expect).
+    #[tokio::test]
+    async fn test_updateable_server() {
+        setup_tracing_subscriber();
+
+        let mut dns_server = DnsServerBuilder::new("example.com", "test.example.com")
+            .run_updateable()
+            .await;
+
+        let service = service::Name("test.example.com".into());
+        let client = Client::new(
+            &DnsResolverConfig::default(),
+            dns_server.address,
+            Duration::from_millis(100),
+        );
+        match client.lookup_socket_v6(&service).await {
+            Err(err) => match err.kind() {
+                ResolveErrorKind::NoRecordsFound { .. } => (),
+                _ => panic!("unexpected error: {err}"),
+            },
+            Ok(backends) => panic!("unexpectedly found {} backends", backends.len()),
+        }
+
+        dns_server
+            .add_backend(1234, "test001.example.com.", 1000)
+            .await;
+        let backends = client
+            .lookup_socket_v6(&service)
+            .await
+            .expect("successful lookup");
+        assert_eq!(backends.len(), 1);
+
+        dns_server
+            .add_backend(1235, "test002.example.com.", 1000)
+            .await;
+        let backends = client
+            .lookup_socket_v6(&service)
+            .await
+            .expect("successful lookup");
+        assert_eq!(backends.len(), 2);
+
+        dns_server.remove_backend("test001.example.com.").await;
+        let backends = client
+            .lookup_socket_v6(&service)
+            .await
+            .expect("successful lookup");
+        assert_eq!(backends.len(), 1);
+    }
+
+    #[tokio::test]
+    async fn test_no_records_found() {
+        setup_tracing_subscriber();
+
+        // Start a DNS server that reports no records for the service.
```

Comment:
It's reporting a single record for the service, right? not "no records"

Reply:
Oops, yeah, this is leftover from an earlier version of the test where I did start it with no records. Will fix.